writerseeker.go

```diff
@@ -48,7 +58,7 @@ func (ws *WriterSeeker) Seek(offset int64, whence int) (int64, error) {
 
 // Reader returns an io.Reader. Use it, for example, with io.Copy, to copy the content of the WriterSeeker buffer to an io.Writer
 func (ws *WriterSeeker) Reader() io.Reader {
-	return bytes.NewReader(ws.buf)
+	return bytes.NewReader(ws.buf.Bytes())
```

Comment:
One potential issue is that the readers returned by this method and `BytesReader` are only usable until the next call to `Write` (since it could trigger a reallocation). I think that this is already an issue in the original code, though. Should it be documented?